axolotls/string_column.py

```diff
@@ -0,0 +1,102 @@
+import string
+from .column_base import ColumnBase
+from . import dtypes as dt
+from typing import Optional
+import torch
+
+class StringColumn(ColumnBase):
+    def __init__(
+        self, 
+        values: torch.ByteTensor, 
+        offsets: torch.Tensor,
+    ) -> None:
+        # String Column doesn't support null value for now
+        # It is straightforward to support it (with extra optional), but hold on for simplicity, 
+        #   and only add it once there is clear use cases.
+        super().__init__(dtype=dt.string)
+
+        if not isinstance(values, torch.ByteTensor) or values.dim() != 1:
+            raise ValueError("StringColumn expects values to be 1D ByteTensor")
+        if not isinstance(offsets, torch.Tensor) or offsets.dim() != 1 or offsets.dtype != torch.int32:
+            raise ValueError("StringColumn expects 1D offsets int32 Tensor")
+
+        self._values = values
+        self._offsets = offsets
+
+    def clone(self) -> "StringColumn":
+        return StringColumn(
+            values=self.values.detach().clone(),
+            offsets=self.offsets.detach().clone(),
+        )
+
+    def __getitem__(self, key) -> str:
+        if isinstance(key, int):
+            bytes: torch.ByteTensor = self.values[self.offsets[key] : self.offsets[key + 1]]
+            return bytearray(bytes).decode('utf-8')
```

Comment:
@dracifer 

> wonder if we should want to print or have a utility func to print the stored strings instead of having to convert to arrow to see what is the value.

Indeed, this is done by here ~

```python
>>> import pyarrow as pa
>>> import axolotls as ax
>>> arr1 = pa.array(["abc", "de", "XYZ", "不是ascii"])
>>> col1 = ax.StringColumn.from_arrow(arr1)

>>> col1[2]
'XYZ'
>>> col1[3]
'不是ascii'

>>> col1
0  abc
1  de
2  XYZ
3  不是ascii
dtype: string, length: 4
```